From 2b599164f3c7622ffd04d5351591c530fc07a965 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:31 -0400
Subject: [usb] sierra: minimise no-suspend window during close

Message-id: <1410887463-20674-440-git-send-email-dzickus@redhat.com>
Patchwork-id: 95456
O-Subject: [RHEL7 PATCH 439/591] USB: sierra: minimise no-suspend window during close
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 0287d5c5cda82ebe87e79e5a298f40f0ca05e5ef
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:08 2014 +0200

    USB: sierra: minimise no-suspend window during close

    Move usb_autopm_get_interface_no_resume to the end of close(). This
    makes the window during which suspend is prevented before the final put
    in USB serial core slightly smaller.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index 9673319..be4a759 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -769,8 +769,6 @@ static void sierra_close(struct usb_serial_port *port)
 
  portdata = usb_get_serial_port_data(port);
 
- usb_autopm_get_interface_no_resume(serial->interface);
-
  spin_lock_irq(&intfdata->susp_lock);
  portdata->opened = 0;
  if (--intfdata->open_ports == 0)
@@ -796,6 +794,8 @@ static void sierra_close(struct usb_serial_port *port)
   sierra_release_urb(portdata->in_urbs[i]);
   portdata->in_urbs[i] = NULL;
  }
+
+ usb_autopm_get_interface_no_resume(serial->interface);
 }
 
 static int sierra_open(struct tty_struct *tty, struct usb_serial_port *port)
-- 
1.7.1